bycycle/tests/cyclepoints/test_extrema.py

```diff
@@ -37,12 +37,12 @@ def test_find_extrema(first_extrema):
     minima = argrelextrema(sig, np.less)
 
     # Find peaks and troughs using bycycle and make sure match scipy
-    peaks, troughs = find_extrema(sig, fs, f_range, boundary=1, first_extrema=first_extrema)
+    peaks, troughs = find_extrema(sig, fs, f_range, first_extrema=first_extrema)
```

Comment:
The simulated signal used in this test is a special case and this won't pass for other signals. `argrelextrema` + sig can find the ~~similar~~ exact same (`np.testing.assert_equal` will also pass) maximum/minima as `find_extrema` + sig_filt. The only difference is the scipy func returns two extra values, noted in your next comment (I'll need to look into this). `argrelextrema` will also not produce the same maxima/minima (as `find_extrema`) when the same filtered signal is used that is created within `find_extrema`,

For your second point, `find_extrema` will not work if only `sig_filt` is passed in. Rise/Decay points are found with sig_filt, but finding peaks and troughs requires the orginal signal:
```
sig_pad[int(filt_len):] = sig
# ...
peaks[p_idx] = np.argmax(sig_pad[last_rise:next_decay]) + last_rise
# ...
roughs[t_idx] = np.argmin(sig_pad[last_decay:next_rise]) + last_decay
```
I suppose `sig` and `sig_filt` could both be passed in, requiring `sig_filt` to be created prior to each `find_extrema` call. However, filtering can be turned off in the function (as is) using `filter_kwargs=None` and  `pad=False`, and full filtering flexibility/control could be achieved by adding `pass_type`, `remove_edges` to `find_extrema`.

Reply:
`pass_type` has been added as an optional kwarg. I didn't include `remove_edges` since this has to be done when padding anyways (it's actually done by adjusting `peaks` and `troughs` - the reason why is noted in a different comment.